app/src/standard/java/at/bitfire/icsdroid/ui/DonateDialogService.kt

```diff
@@ -48,55 +52,32 @@ class DonateDialogService: ComposableStartupService {
         const val SHOW_EVERY_MILLIS_DISMISS = ONE_DAY_MILLIS * 14
     }
 
-    private var preferences: SharedPreferences? = null
-
-    override fun initialize(activity: AppCompatActivity) {
-        if (preferences != null) return
-        preferences = getPreferences(activity)
-    }
-
-    // Note: Workaround for Android 6 (API 23) devices with low memory (1GB)
-    private fun getPreferences(activity: AppCompatActivity): SharedPreferences =
-        preferences ?: activity.getPreferences(0).also { preferences = it }
+    override val flags: Int = FLAG_DONATION_DIALOG
```

Comment:
Why plural? It can only ever hold one flag at a time, no?

Reply:
We can hold multiple flags with int values, it uses logical binary operations for that. In this case, `0` would be no flags. If we apply `1` using `OR` we are setting the first bit to `1`. The next possible value would be `2` (`b10`), the next one `4` (`b100`) and so on. This way, we can store in one variable multiple flags.

For example, right now we have the `FLAG_DONATION_DIALOG` flag, which is `1`. But let's say at some point we have `FLAG_B`, which has value `2`.

We can store `FLAG_DONATION_DIALOG OR FLAG_B`w, which would be in binary `b01+b10=b11`. In this case the flag decimal value would be `3`, but for us, it would mean that it has two flags.

If we ever want to check whether the flag `FLAG_B` in stored, we can simply apply the operation `b11` (the value stored) `AND` `FLAG_B`:
```
11 AND 10 = 10
```
Since it's equal to `FLAG_B`, it means that the flag is present. Otherwise, let's say we have `b01` stored:
```
01 AND 10 = 0
```
which is not equal to `FLAG_B`, meaning the flag is not present.

_Btw I have just realized that the operation is not correct [here](https://github.com/bitfireAT/icsx5/pull/371/files#diff-298c775d0a64e9b72de9a71953b9b45140628c0cf69083af0e021e2081293b36R77)_

Reply:
Got it. Thanks for the explanation. It looked uneccessarily complex at first. But I guess once you know what it does it's not that bad and we seem do the same thing it DAVx5 already :+1:

Reply:
I would still like a comment on the flag variable though. [Like in DAVx5](https://github.com/bitfireAT/davx5-ose/blob/62d5a21d05ff5631a71e153c4febf9ad29cf3cad/app/src/main/kotlin/at/bitfire/davdroid/resource/LocalResource.kt#L40).